x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/index.tsx

```diff
@@ -123,7 +184,51 @@ const AttackDiscoveryPageComponent: React.FC = () => {
 
   const pageTitle = useMemo(() => <PageTitle />, []);
 
-  const onGenerate = useCallback(async () => fetchAttackDiscoveries(), [fetchAttackDiscoveries]);
+  const { indexPattern } = useSourcererDataView();
+
+  // filterQuery is the combined search bar query and filters in ES format:
+  const [filterQuery, kqlError] = useMemo(
+    () =>
+      convertToBuildEsQuery({
+        config: getEsQueryConfig(uiSettings),
+        indexPattern,
+        queries: [query ?? getDefaultQuery()], // <-- search bar query
+        filters: filters ?? [], // <-- search bar filters
+      }),
+    [filters, indexPattern, query, uiSettings]
```

Comment:
In `main`, the `convertToBuildEsQuery` function called here accepts a parameter named `dataViewSpec`, which is typed as:

```typescript
dataViewSpec: DataViewSpec | undefined;
```

In `8.x`, the `convertToBuildEsQuery` function takes a different parameter named `indexPattern` instead of `dataViewSpec`, and it's typed differently:

```typescript
indexPattern: DataViewBase | undefined;
```

The page was updated to call the `convertToBuildEsQuery` function using the `8.x` version of the parameter.